File: alvara/app/Models/Application.php
<?php

namespace App\Models;

use App\States\ApplicationStateOpen;
use App\States\ApplicationStateRejected;
use App\States\ApplicationStateSended;
use App\States\ApplicationStateSuccess;
use App\States\IApplicationState;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Enums\ApplicationStatus;
use App\Enums\ApplicationLicenseType;
use PHPUnit\Exception;

class Application extends Model
{
    use HasFactory;

    protected $fillable = [
        "id",
        "status",
        "title",
        "type",
        "company_id",
    ];

    protected $cast = [
        "status" => ApplicationStatus::class,
        "type" => ApplicationLicenseType::class,
    ];
    public function attachments()
    {
        return $this->hasMany(Attachment::class);
    }

    public function getStatusController(): IApplicationState
    {
        switch ($this->status) {
            case ApplicationStatus::SUCCESS->value:
                return new ApplicationStateSuccess($this);
            case ApplicationStatus::OPEN->value:
                return new ApplicationStateOpen($this);
            case ApplicationStatus::SENDED->value:
                return new ApplicationStateSended($this);
            case ApplicationStatus::REJECTED->value:
                return new ApplicationStateRejected($this);

            default:
                throw new Exception("Status não encontrado");
                break;
        }
    }
}

File: app/lib/domain/models/Application.dart
import 'package:app/domain/models/IModel.dart';
import 'package:app/data/models/Application.dart' as api;

class Application implements IModel {
  @override
  int? id;
  String? status;
  String title;
  String type;
  int companyId;

  Application({
    this.id,
    this.status,
    required this.title,
    required this.type,
    required this.companyId,
  });

  factory Application.fromApi(api.Application object) {
    return Application(
      id: object.id,
      status: object.status,
      title: object.title,
      type: object.type,
      companyId: object.companyId,
    );
  }
  @override
  api.Application toApi() {
    return api.Application(
      status: status,
      title: title,
      type: type,
      companyId: companyId,
    );
  }
}
